### ISSUE TITLE
Split packages: depends and makedepends for the build should be used from base

### ISSUE THREAD

**Issue by bertptrs** [actionless/pikaur#320]
```
pikaur -Vq
Pikaur v1.3.2
Pacman v5.1.1 - libalpm v11.0.1
```

Following [a discussion on the aur-general mailing](https://lists.archlinux.org/pipermail/aur-general/2018-December/034683.html) list, it was determined that sometimes pikaur determines build dependencies incorrectly. The package in question has been removed (since there were some other issues with it) but I've uploaded the [the original PKGBUILD](https://gist.github.com/bertptrs/f3b67701bdbde2ac2c334ece40c0e39a) as a gist.

This particular package is a split package, and has dependencies listed globally as well as in the individual packages. When pikaur determines the dependencies to install, it will take the depends for the particular package you've chosen (either the python- or python2 version) instead of the global depends.

After that, `makepkg` will be started to build the package, and then complain that not all dependencies can be resolved because some of the dependencies need to come from the AUR.

While we can agree that what the package is doing is weird, correct behaviour is to install the dependencies from the pkgbase rather than from the split package, since that is what `makepkg` expects.

**Comment by eli-schwartz** [comment]
> The package in question has been removed (since there were some other issues with it)

It is still there...

**Comment by actionless** [comment]
I can't reproduce it in the way how you describing it, the packages are built successfully but can't be installed together:

```
sudo pacman --color=always --upgrade /home/lie/.cache/pikaur/pkg/python-magic-wormhole-0.11.2-1-any.pkg.tar /home/lie/.cache/pikaur/pkg/python2-magic-wormhole-0.11.2-1-any.pkg.tar
[sudo] password for lie:
loading packages...
resolving dependencies...
looking for conflicting packages...
error: unresolvable package conflicts detected
error: failed to prepare transaction (conflicting dependencies)
:: python-magic-wormhole and python2-magic-wormhole are in conflict
```

**Comment by actionless** [comment]
I can't reproduce it in the way how you describing it, the packages are built successfully but can't be installed together:

```
sudo pacman --color=always --upgrade /home/lie/.cache/pikaur/pkg/python-magic-wormhole-0.11.2-1-any.pkg.tar /home/lie/.cache/pikaur/pkg/python2-magic-wormhole-0.11.2-1-any.pkg.tar
loading packages...
resolving dependencies...
looking for conflicting packages...
error: unresolvable package conflicts detected
error: failed to prepare transaction (conflicting dependencies)
:: python-magic-wormhole and python2-magic-wormhole are in conflict
```

**Comment by actionless** [comment]
and please don't ignore issue template, otherwise i need to guess what exactly did you did, and reading human text is less useful than output

**Comment by bertptrs** [comment]
Log files are here: https://ptpb.pw/u2KR

**Comment by actionless** [comment]
i still don't know which command did you passed

**Comment by bertptrs** [comment]
Added that in the comment, after I noticed it was missing from the log…

**Comment by actionless** [comment]
it works for me:  [python-magic-wormhole.log](https://github.com/actionless/pikaur/files/2672520/python-magic-wormhole.log)

do you have python2-magic-wormhole already installed?

**Comment by bertptrs** [comment]
I don't. It might work if I did, since then I would have the python2 dependencies. Then I just won't be able to install it.

**Comment by actionless** [comment]
so far i don't see any problem with pikaur -- split package have conflicting dependencies -- that's it

**Comment by bertptrs** [comment]
The problem is that the depends are overwritten in each separate `package_` function, but for the building it should look at the global `depends` and `makedepends` it shouldn't look at that, it should install the from the global depends. The dependencies listed in the `package_` functions are only for installing, not building.

**Comment by actionless** [comment]
i see

**Comment by actionless** [comment]
i've pushed a possible fix, please check with `pikaur-git`

**Comment by actionless** [comment]
1) run the same pikaur command command with `--debug` flag
2) try running the same makepkg/pacman commands as pikaur logging -- but manually

**Comment by actionless** [comment]
i've double-checked PKGBUILD and i've got to the same conclusion as at first (https://github.com/actionless/pikaur/issues/320#issuecomment-446633753):
 
built packages are failing to install with pacman because of these lines[1](https://gist.github.com/bertptrs/f3b67701bdbde2ac2c334ece40c0e39a#file-pkgbuild-L15-L17)

but thanks for finding another bug with handling build-deps of split packages while investigating this anyway :)

**Comment by actionless** [comment]
found one more problem on this topic, neither AUR RPC nor Web UI advertising those build dependencies packages:

https://aur.archlinux.org/rpc/?v=5&type=info&arg[]=python-magic-wormhole
https://aur.archlinux.org/packages/python-magic-wormhole/

that means what for every AUR package list of build dependencies should be recomputed after downloading them (recursively)

**Comment by bertptrs** [comment]
I don't know if you'd still like my debug log files, but here they are.

```
pikaur -Vq
Pikaur v1.3.3.r2.g09987eb
Pacman v5.1.1 - libalpm v11.0.1
```

`pikaur -S --debug python-magic-wormhole`: [full log](https://ptpb.pw/wtoG)

**Comment by actionless** [comment]
@bertptrs thanks for your assistance, i've tried to workaround missing info from RPC in a way how i described in a previous comment and also wrote a test for this package, please check also with `pikaur-git`

**Comment by actionless** [comment]
@bertptrs could you please confirm it's working as expected for you?
in that case i'll make a new release with this fix

**Comment by actionless** [comment]
@bertptrs ping

**Comment by bertptrs** [comment]
Pong. I can confirm that it works now. I imagine this was not the prettiest fix.

**Comment by actionless** [comment]
do you have any suggestions?
as i mentioned previously those extra packages are not advertised in neither  AUR RPC nor its Web UI. so it's possible to know them only after cloning package repo from AUR

**Comment by bertptrs** [comment]
Maybe it would be good to submit this as a change to the AUR RPC? It probably was never intended like this, but this _is_ how makepkg interprets it.

It is entirely possible that this has never occurred before since most people will add those build packages to `makedepends` instead. Either way thanks for resolving this.

**Comment by actionless** [comment]
fixing AUR RPC it's obviously out of scope of pikaur issue tracker, and just FYI, the previous fix for RPC (search by Provides package field) is not merged yet so i'm lost interest in trying to make it work fine and instead i am just trying to make some workarounds on my side to make the users less affected by those problems

**Comment by svitoz** [comment]
> Alad is continuing overindulging her arch wiki privileges

I assure you that's not the case here because you are first to test other helpers and conclude yay also does not handle it well. I could have made same mistake and that would not have been abuse like it wasn't here. Again mistake here is not what you are suggesting but because it is not reflected in criteria example packages on the wiki against which one should test against.

Alad suggested to make test environment that automatically test all the helpers and updates the table. I think that would solve the issue quite well for everybody. Without any test suite mistakes will be made. I will poke around this direction but don't expect much, I am very lazy.

Another issue where you are not helping is you calling polygamma Alad's **special** friend which literally means retarded, then stating that making fun of Alad is totally fine and marking Alad's comment **disruptive** when yours is merely **off-topic**, even though you actually making direct insults where Alad was making a wicked joke. This puts you in a weak position and you certainly can't call out somebody else from there, especially in relation to personality traits and being fair.

**Comment by actionless** [comment]
btw test for the newly-found split package problem is also added, but to a different testsuite, https://github.com/actionless/pikaur/blob/master/pikaur_test/test_regressions.py#L82-L91 since such case is not part of arch wiki criterias
